src/Symfony/Component/Form/Extension/Core/Type/EnumType.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Form\Extension\Core\Type;
+
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\Form\Exception\RuntimeException;
+use Symfony\Component\OptionsResolver\Options;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+/**
+ * A choice type for native PHP enums.
+ *
+ * @author Alexander M. Turek <me@derrabus.de>
+ */
+class EnumType extends AbstractType
+{
+    public function configureOptions(OptionsResolver $resolver): void
+    {
+        $resolver->setRequired(['class']);
+
+        $resolver->addNormalizer('class', static function (Options $options, $class): string {
```

Comment:
this validation should be handled by `setAllowedTypes` and `setAllowedValues` instead of abusing a normalizer for that. This will also make that validation visible when using the `debug:form` command

Reply:
Of course. 🙈  Fixed!